rest/processGlobalQueue.ts

```diff
@@ -118,18 +118,33 @@ export async function processGlobalQueue(rest: RestManager) {
 
         // If NOT rate limited remove from queue
         if (response.status !== 429) {
+          let json = undefined;
           if (response.type) {
-            console.log(JSON.stringify(await response.json()));
+            json = await response.json();
+            console.log(JSON.stringify(json));
           }
 
-          request.request.reject(new Error(`[${response.status}] ${error}`));
+          request.request.reject({
+            ok: false,
+            status: response.status,
+            error: {
+              raw: new Error(`[${response.status}] ${error}`),
+              name: error,
+            },
+            body: json ? JSON.stringify(json) : undefined,
+          });
         } else {
           if (request.payload.retryCount++ >= rest.maxRetryCount) {
             rest.debug(`[REST - RetriesMaxed] ${JSON.stringify(request.payload)}`);
             // REMOVE ITEM FROM QUEUE TO PREVENT RETRY
-            request.request.reject(
-              new Error(`[${response.status}] The request was rate limited and it maxed out the retries limit.`),
-            );
+            request.request.reject({
+              ok: false,
+              status: 429,
+              error: {
+                raw: new Error(`[${response.status}] The request was rate limited and it maxed out the retries limit.`),
+                name: "The request was rate limited and it maxed out the retries limit.",
+              },
```

Comment:
Completely remove `new Error()` and make the `runMethod` function construct it.

```suggestion
              error: "The request was rate limited and it maxed out the retries limit.",
```